workitem/workitem_repository.go

```diff
@@ -1193,11 +1193,28 @@ func (r *GormWorkItemRepository) ChangeWorkItemType(ctx context.Context, wiStora
 			// Try to assign the old value to the new field
 			_, err := newField.Type.ConvertToModel(wiStorage.Fields[oldFieldName])
 			if err != nil {
-				// The field might be a list type. Try to assign it to a list
+				// if the new type is a list, stuff the old value in a list and
+				// try to assign it
 				if newField.Type.GetKind() == KindList {
-					interfaceArray, ok := wiStorage.Fields[oldFieldName].([]interface{})
-					if ok {
-						_, err = newField.Type.ConvertToModel(interfaceArray)
+					var convertedValue interface{}
+					convertedValue, err = newField.Type.ConvertToModel([]interface{}{wiStorage.Fields[oldFieldName]})
```

Comment:
@kwk Why did you cast the value here 
`[]interface{}{wiStorage.Fields[oldFieldName]}` ?
I had done a type assertion in my code.
So the question is when do we type assert and when do we type cast?

Reply:
Based on https://jimsaunders.net/2015/04/23/type-conversion-vs-type-assertion-in-go.html

```go
convertedValue, err = newField.Type.ConvertToModel([]interface{}{wiStorage.Fields[oldFieldName]})
```
should be
```go
 a, _ := wiStorage.Fields[oldFieldName].([]interface{})
convertedValue, err = newField.Type.ConvertToModel(a)
```

Reply:
@jarifibrahim No it shouldn't. And I'm not doing a cast here. I'm creating a new array of interface objects and stuff the old value in it so it becomes a list. This is useful when we want to convert from a field that was a single value field to a field that is a list. I've updated the tests now (692cce4) to include negative tests as well that check if the description is properly updated if a field couldn't be converted. If all passes I'm fine with this PR.

Reply:
@kwk I don't completely understand your answer but we can deal with it later. I am fine with your changes. Thank you :)

Reply:
>Why did you cast the value here
[]interface{}{wiStorage.Fields[oldFieldName]} ?

I thought this is type casting but you're creating a new array here. This makes sense now.
I got confused between type casting and type assertion but this is creating a new array.